src/components/ui/Image/useImage.ts

```diff
@@ -0,0 +1,44 @@
+import { useMemo } from 'react'
+import type { ImgHTMLAttributes } from 'react'
+
+import { urlBuilder } from './thumborUrlBuilder'
+import type { ThumborOptions } from './thumborUrlBuilder'
+
+export interface ImageOptions extends ImgHTMLAttributes<HTMLImageElement> {
+  src: string
+  width: number
+  height: number
+  options?: ThumborOptions
+}
+
+const FACTORS = [1, 2, 3]
+
+export const useImage = ({
+  src: baseUrl,
+  width,
+  height,
+  options = {},
+  ...rest
+}: ImageOptions): ImgHTMLAttributes<HTMLImageElement> => {
+  const { srcSet, src } = useMemo(() => {
+    const builder = urlBuilder(baseUrl, options)
+
+    const srcs = FACTORS.map(
+      (factor) =>
+        `${builder(width * factor, height * factor)} ${width * factor}w`
+    )
```

Comment:
What do you think instead of multiplying by a hardcoded factor, the user set breakpoints like others images components?